src/inout.f90

```diff
@@ -378,82 +465,37 @@ END SUBROUTINE WriteOutputEVal
-     IWidth, CubeDis, LiebDis, CubeConPot, LiebConPot, &
-     PreSeed, str, IErr)
+SUBROUTINE WriteOutputEVec( Inum, NEVals, Lsize, VECS, VECS_size, &
+     IWidth, PreSeed, DirName, MiddleName, IErr)
 
   USE MyNumbers
   USE IChannels
-!  USE DPara
-!  USE IPara
+  USE DPara
+  USE IPara
 
-  INTEGER(KIND=IKIND) Dim, Nx
   INTEGER(KIND=IKIND) Inum, PreSeed, ISSeed, IWidth, IErr, Lsize, VECS_size, NEVals, i
-  REAL(KIND=RKIND) CubeDis, LiebDis, CubeConPot, LiebConPot
 
   REAL(KIND=RKIND) VECS(VECS_size,VECS_size)
 
-  CHARACTER*100 FileName, str
-  CHARACTER*1 SymbolCP,symbolLP
-
-  IF(CubeConPot.GE.0.0D0) Then
-     SymbolCP="+"
-  Else
-     SymbolCP="-"
-  End If
-
-  If(LiebConPot.GE.0.0D0) Then
-     SymbolLP="+"
-  Else
-     SymbolLP="-"
-  End If
-
-
+  CHARACTER(len=*) DirName, MiddleName
+  CHARACTER*50 Prefix, Postfix
+  CHARACTER*200 FileName
   
 !!$  PRINT*,"DBG: WriteOutputEvec()"
-
   IErr= 0
 
-  !   WRITE out the input parameter
-  WRITE(FileName, '(A5,A1,I1,I1,A2,I4.4,A6,A3,A1,I6.6,A3,I6.6,A3,A1,I6.6,A3,I6.6,A2,I5.5,A2,I6.6,A4)') &
-       "Evec_", "L", Dim, Nx, &
-       "_M",IWidth, &
-       "_Espec", &! NINT(100.*ABS(Energy)), &
-       "_CP",SymbolCP,NINT(100.*ABS(CubeConPot)), &
-       "_CD", NINT(100.*ABS(CubeDis)), &
-       "_LP", SymbolLP,NINT(100.*ABS(LiebConPot)), &
-       "_LD", NINT(100.*ABS(LiebDis)), "-c",& 
-       PreSeed, "-N", Inum, ".raw" !"_s", ISSeed, 
-
-!!$  IF(Energy.GE.0.0D0) THEN
-!!$     WRITE(FileName, '(A5,A1,I1,I1,A2,I4.4,A6,A3,I6.6,A3,I6.6,A6,I6.6,A6,I6.6,A2,I5.5,A2,I4.4,A4)') &
-!!$     !WRITE(FileName, '(A5,A1,I1,I1,A2,I4.4,A2,A5,I9.9,A7,I7.7,A7,I7.7,A2,I4.4,A1,I5.5,A4)') &
-!!$          "Evec-","L", Dim, Nx, &
-!!$          "-M", IWidth, &
-!!$          "-Espec", & !NINT(100.0D0*ABS(Energy)), &
-!!$          "-hD", NINT(100.0D0*ABS(CubeDis)), &
-!!$          "-rD", NINT(100.0D0*ABS(LiebDis)), &
-!!$          "-CP", NINT(100.*ABS(CubeConPot)), &
-!!$          "-LP", NINT(100.*ABS(LiebConPot)), &
-!!$          "-c", PreSeed, "-N", Inum, & !"_s", ISSeed, 
-!!$          ".raw"
-!!$  ELSE
-!!$     WRITE(FileName, '(A5,A1,I1,I1,A2,I4.4,A6,A3,I6.6,A3,I6.6,A6,I6.6,A6,I6.6,A2,I5.5,A2,I4.4,A4)') &
-!!$     !WRITE(FileName, '(A5,A1,I1,I1,A2,I4.4,A2,A5,I9.9,A7,I7.7,A7,I7.7,A2,I4.4,A1,I5.5,A4)') &
-!!$          "Evec-","L",Dim, Nx, &
-!!$          "-M", IWidth, &
-!!$          "-Espec", & !NINT(100.0D0*ABS(Energy)), &
-!!$          "-dD", NINT(100.0D0*ABS(CubeDis)), &
-!!$          "-rD", NINT(100.0D0*ABS(LiebDis)), &
-!!$          "-CP", NINT(100.*ABS(CubeConPot)), &
-!!$          "-LP", NINT(100.*ABS(LiebConPot)), &
-!!$          "-c", PreSeed, "-N", Inum, & !"_s", ISSeed, 
-!!$          ".raw"
-!!$  ENDIF
-
-!!$  PRINT*, "WriteOutputEVec(): ", FileName
-
-  OPEN(UNIT= IChEVec, ERR= 40, STATUS= 'UNKNOWN', FILE=TRIM(ADJUSTL(str))//"/"//FileName)
+  WRITE(Prefix, '(A5)') "Evec_"
+  WRITE(Postfix, '(A2,I5.5A2,I5.5,A4)') "-c",  PreSeed, "-N", Inum, ".raw" 
```

Comment:
missed a comma in '(A2,I5.5**,**A2,I5.5,A4)')

Reply:
Thanks, that was indeed a stupid mistake. Well done for spotting it, would have caused havoc with filenaming when used in production. Now corrected in https://github.com/DisQS/LiebExactDiag/pull/15/commits/e114ede6aef279ea7fac07be05114cc199646dd3.